src/main/native/org_bblfsh_client_v2_libuast_Libuast.cc

```diff
@@ -0,0 +1,136 @@
+#include "org_bblfsh_client_v2_libuast_Libuast.h"
+#include "jni_handle.h"
+#include "jni_utils.h"
+#include "org_bblfsh_client_v2_Context.h"
+
+#include "libuast.h"
+#include "libuast.hpp"
+
+JavaVM *jvm;  // FIXME(bzz): double-check and document
+
+jobject asJvmBuffer(uast::Buffer buf) {
```

Comment:
Here and throughout, please consider wrapping utility code required only by the current translation unit in anonymous namespace(s).